### ISSUE TITLE
TautomerCanonicalizer gives unexpected/forbidden form of phosphoric acid

### ISSUE THREAD

**Issue by benbowen** [mcs07/MolVS#20]
I'm converting all the molecules in my database to canonical-tautomers and noticed that things like NADH looked weird.  You can see it most plainly for phosphoric acid.  I didn't expect the Hydrogen on the phosphorous.  Is this the correct/expected behavior?

```python
from rdkit import Chem
from rdkit.Chem import Draw
from molvs.tautomer import TautomerCanonicalizer

original_smiles = 'OP(=O)(O)O'

original_mol = Chem.MolFromSmiles(original_smiles)
tautomerized_mol = TautomerCanonicalizer().canonicalize(original_mol)

Draw.MolsToGridImage([original_mol,tautomerized_mol],
                     molsPerRow=3,subImgSize=(200,200),
                     legends=['original','tautomer'])
```

![image](https://user-images.githubusercontent.com/810953/35994767-4a5e3038-0cc6-11e8-9f03-d97fc5109515.png)

**Comment by benbowen** [comment]
NADH looks like this

```python
original_smiles = 'NC(=O)C1=CN([C@@H]2O[C@H](COP(=O)(O)OP(=O)(O)OC[C@H]3O[C@@H](N4C=NC5=C4N=CN=C5N)[C@H](O)[C@@H]3O)[C@@H](O)[C@H]2O)C=CC1'

original_mol = Chem.MolFromSmiles(original_smiles)
tautomerized_mol = TautomerCanonicalizer().canonicalize(original_mol)

Draw.MolsToGridImage([original_mol,tautomerized_mol],
                     molsPerRow=1,subImgSize=(600,300),
                     legends=['original','tautomer'])
```

![image](https://user-images.githubusercontent.com/810953/35995032-0ef80676-0cc7-11e8-812e-37c724041535.png)

**Comment by mcs07** [comment]
I think this is caused by the phosphonic acid rules: https://github.com/mcs07/MolVS/blob/master/molvs/tautomer.py#L130

It can probably be fixed by making the SMARTS pattern more strict to match only the intended target:
https://en.wikipedia.org/wiki/Phosphorous_acid

**Comment by benbowen** [comment]
You are correct, removing that rule stops that moiety from being modified.  When you say, "more strict", you think specify an explicit number of bonds on the Phosphorous in the SMARTS pattern?

Why does rdkit allow 6 bonds on the phosphorous? Rdkit is a vast package, but looking at the definition of Phosphorous, it has[ max bonds of 5](https://github.com/rdkit/rdkit-orig/blob/master/rdkit/Chem/PeriodicTable.py#L37).

If I do SantizeMol, the hydrogen stays put.  When I paste the structure into ChemDraw, its not valid.
